changelog.d/11616.misc

```diff
@@ -0,0 +1 @@
+Allow specifying the application service-specific `user_id` parameter in the `join` test helper.
```

Comment:
Should this have the same changelog as #11615?

Reply:
I don't really know, truthfully

Reply:
I think it should, something like "Refactor test helpers for application services." seems fine.

Ideally if the actual changes land in the same release we can reword them there to point to the actual improvement / bugfix.